core/templates/pages/admin-page/platform-parameters-tab/admin-platform-parameters-tab.component.spec.ts

```diff
@@ -0,0 +1,668 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests for Admin Platform Parameters tab component.
+ */
+
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { ComponentFixture, fakeAsync, async, TestBed, flushMicrotasks, tick } from
+  '@angular/core/testing';
+import { FormsModule } from '@angular/forms';
+import cloneDeep from 'lodash/cloneDeep';
+
+import { AdminPageData } from 'domain/admin/admin-backend-api.service';
+import { AdminDataService } from 'pages/admin-page/services/admin-data.service';
+import { AdminTaskManagerService } from
+  'pages/admin-page/services/admin-task-manager.service';
+import { PlatformFeatureAdminBackendApiService } from
+  'domain/platform_feature/platform-feature-admin-backend-api.service';
+import { AdminPlatformParametersTabComponent } from
+  // eslint-disable-next-line max-len
+  'pages/admin-page/platform-parameters-tab/admin-platform-parameters-tab.component';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { PlatformParameterFilterType, ServerMode } from
+  'domain/platform_feature/platform-parameter-filter.model';
+import { FeatureStage, PlatformParameter } from 'domain/platform_feature/platform-parameter.model';
+import { HttpErrorResponse } from '@angular/common/http';
+
+
+describe('Admin page platform parameters tab', () => {
+  let component: AdminPlatformParametersTabComponent;
+  let fixture: ComponentFixture<AdminPlatformParametersTabComponent>;
+  let adminDataService: AdminDataService;
+  let featureApiService: PlatformFeatureAdminBackendApiService;
+  let adminTaskManagerService: AdminTaskManagerService;
+  let windowRef: WindowRef;
+
+  let updateApiSpy: jasmine.Spy;
+
+  let mockConfirmResult: (val: boolean) => void;
+  let mockPromptResult: (msg: string | null) => void;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [FormsModule, HttpClientTestingModule],
+      declarations: [AdminPlatformParametersTabComponent],
+      providers: [AdminTaskManagerService],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(AdminPlatformParametersTabComponent);
+    component = fixture.componentInstance;
+    adminDataService = TestBed.get(AdminDataService);
+    featureApiService = TestBed.get(PlatformFeatureAdminBackendApiService);
+    windowRef = TestBed.get(WindowRef);
+    adminTaskManagerService = TestBed.get(AdminTaskManagerService);
+
+    let confirmResult = true;
+    let promptResult: string | null = 'mock msg';
+    spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
+      confirm: () => confirmResult,
+      prompt: () => promptResult,
+      alert: () => null
+    } as unknown as Window);
```

Comment:
Can you mock this instead?

Reply:
Hey @gp201, can you please elaborate on it more? We are mocking the `nativeWindow` here, right?
Also, when I took a look at other test files we are doing something similar to this in order to mock the `nativeWindow`, also main reason why I did this is that we use a similar way to mock `nativeWindow` for the `features` component as well. Let me know what do you think?
Thanks

Reply:
@lkbhitesh07 Sorry, I was not clear. I was requesting you to mock windowref instead.

example:
https://github.com/oppia/oppia/blob/3aab4e534cf5743dd8f60a4dadebb8f698af270c/core/templates/i18n/i18n.service.spec.ts#L42-L62

Reply:
Got it, Done with the changes.
Thanks